lib/message_builder.rb

```diff
@@ -94,11 +98,85 @@ def no_pull_requests
     render "no_pull_requests"
   end
 
+  def prs_count_message
+    if @all_prs_count > 1
+      "You have #{@all_prs_count} Dependabot PRs open on the following apps:"
+    else
+      "You have 1 Dependabot PR open on the following app:"
+    end
+  end
+
+  def repo_details(repo)
+    pr_count = repo[:pr_count]
+
+    if pr_count > 1
+      "oldest: #{days_plural(repo[:oldest_pr])}, newest: #{days_plural(repo[:newest_pr])})"
+    else
+      "opened #{days_plural(repo[:oldest_pr])})"
+    end
+  end
+
+  def github_api_errors_warning
+    "\n\n:warning: #{@github_api_errors} errors while fetching security alerts. :warning:" \
+    "\nPlease check that you have the <#{'https://docs.publishing.service.gov.uk/manual/configure-github-repo.html#when-you-create-a-new-repo'}|right permissions> on all your repos:\n" \
+  end
+
+  def security_alerts_intro
+    message = base_alert_text
+    message += github_api_errors_warning if @github_api_errors.positive?
+    message += security_alerts_count_message if @all_alerts_count.positive? && @security_prs_ordered.any?
```

Comment:
I think we could simplify this to just:

```suggestion
    message += security_alerts_count_message if @security_prs_ordered.any?
```